sql/atxs/atxs.go

```diff
@@ -486,20 +571,29 @@ func LatestN(db sql.Executor, n int) ([]CheckpointAtx, error) {
 		stmt.ColumnBytes(5, catx.SmesherID[:])
 		catx.Sequence = uint64(stmt.ColumnInt64(6))
 		stmt.ColumnBytes(7, catx.Coinbase[:])
+		if sql.IsNull(stmt, 8) {
+			ierr = errors.New("missing nonce")
+			return false
+		} else {
+			catx.VRFNonce = types.VRFPostIndex(stmt.ColumnInt64(8))
+		}
```

Comment:
This will break checkpoint restore - if the ATX was published without a VRF nonce it has to be returned here without one.

Restoring the checkpoint involves running all ATXs of the checkpoint through the ATX handler again. If you changed them (by adding or removing the nonce) they will fail to validate.

Ideally this function should really just read the blobs, decode them and ignore what's in the main atx table.

In a later step we will have to change checkpoint to not even require to decode the ATXs at all.

cc @poszu

Reply:
`catx.VRFNonce` was being populated in the `checkpointDB` function: https://github.com/spacemeshos/go-spacemesh/pull/5791/files#diff-c6ae423644f8c867db6bf09a7d97c28e23ce4a00f2c2fe7a0697739061fb1561L74-L79
by means of `VRFNonce` function which does epoch+pubkey based query
The `nonce` column here doesn't necessarily correspond to the value of `VRFNonce` field in the table, but rather basically is what `VRFNonce` func would have returned for the atxs' `NodeID` and `Epoch+1`.